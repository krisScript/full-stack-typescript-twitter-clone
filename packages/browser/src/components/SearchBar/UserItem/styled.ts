import styled from 'styled-components';

export const UserItemWrapper = styled('li')`
  a {
    width: 100%;
    height: 100%;
    display: grid;
    grid-template-columns: 1fr 7fr;
    align-items: center;
    padding: 0.5rem;
    background: ${props => props.theme.background};
    color: ${props => props.theme.primary};
    border-radius: 0.3rem;
    ${props => props.theme.mixins.border}
  }
`;
export default UserItemWrapper;
