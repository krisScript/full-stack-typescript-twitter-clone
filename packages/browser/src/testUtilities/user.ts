import User from 'types/User';

const user: User = {
  username: 'username',
  handle: 'userHandle',
  email: 'mockUserEmail@mail.com',
  profilePhoto: 'default',
  headerPhoto: 'default',
  date: 'mockDate',
  avatar: 'someImagePublicId',
  cover: 'someImagePublicId',
  followers: 0,
  following: [],
  likes: [],
  bookmarks: [],
  retweets: [],
  replies: [],
  _id: 'id',
};
export default user;
