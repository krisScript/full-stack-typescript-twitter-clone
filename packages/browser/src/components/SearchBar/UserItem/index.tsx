import React, { FC, memo } from 'react';
import { Link } from 'react-router-dom';
import Avatar from 'components/Avatar';
import User from 'types/User';
import { UserItemWrapper } from './styled';

interface UserItemProps {
  user: User;
}
export const UserItem: FC<UserItemProps> = ({ user }) => {
  const { handle, _id } = user;
  return (
    <UserItemWrapper>
      <Link to={`/users/${_id}`} key={_id}>
        <Avatar />
        <span>@{handle}</span>
      </Link>
    </UserItemWrapper>
  );
};
export default memo(UserItem);
