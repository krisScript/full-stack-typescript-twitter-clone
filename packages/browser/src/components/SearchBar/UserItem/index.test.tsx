import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import user from 'testUtilities/user';
import UserItem from './index';

describe('UserItem', () => {
  it('render UserItem', () => {
    expect.assertions(1);

    const { container } = render(<UserItem user={user} />);

    expect(container).toMatchSnapshot();
  });
});
