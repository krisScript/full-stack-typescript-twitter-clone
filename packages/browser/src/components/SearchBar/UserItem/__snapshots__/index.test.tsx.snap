// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem render UserItem 1`] = `
<div>
  <li
    class="sc-AykKE eSxiqy"
  >
    <a
      href="/users/id"
    >
      <div
        class="sc-AykKC bvduzj"
      >
        <img
          alt="avatar"
        />
      </div>
      <span>
        @
        userHandle
      </span>
    </a>
  </li>
</div>
`;
